llrpConnector/src/main/java/io/vantiq/extsrc/llrpConnector/LLRPConnector.java

```diff
@@ -0,0 +1,1546 @@
+import org.llrp.ltk.generated.messages.ENABLE_EVENTS_AND_REPORTS;
+import org.llrp.ltk.generated.messages.ENABLE_ROSPEC;
+import org.llrp.ltk.generated.messages.ENABLE_ROSPEC_RESPONSE;
+import org.llrp.ltk.generated.messages.GET_READER_CAPABILITIES;
+import org.llrp.ltk.generated.messages.GET_READER_CAPABILITIES_RESPONSE;
+import org.llrp.ltk.generated.messages.GET_READER_CONFIG_RESPONSE;
+import org.llrp.ltk.generated.messages.KEEPALIVE_ACK;
+import org.llrp.ltk.generated.messages.GET_READER_CONFIG;
+import org.llrp.ltk.generated.messages.READER_EVENT_NOTIFICATION;
+import org.llrp.ltk.generated.messages.RO_ACCESS_REPORT;
+import org.llrp.ltk.generated.messages.SET_READER_CONFIG;
+import org.llrp.ltk.generated.messages.SET_READER_CONFIG_RESPONSE;
+import org.llrp.ltk.generated.messages.STOP_ROSPEC;
+import org.llrp.ltk.generated.messages.STOP_ROSPEC_RESPONSE;
+import org.llrp.ltk.generated.parameters.AISpec;
+import org.llrp.ltk.generated.parameters.AISpecStopTrigger;
+import org.llrp.ltk.generated.parameters.AccessReportSpec;
+import org.llrp.ltk.generated.parameters.C1G2EPCMemorySelector;
+import org.llrp.ltk.generated.parameters.EventsAndReports;
+import org.llrp.ltk.generated.parameters.EventNotificationState;
+import org.llrp.ltk.generated.parameters.InventoryParameterSpec;
+import org.llrp.ltk.generated.parameters.KeepaliveSpec;
+import org.llrp.ltk.generated.parameters.LLRPStatus;
+import org.llrp.ltk.generated.parameters.ROBoundarySpec;
+import org.llrp.ltk.generated.parameters.ROReportSpec;
+import org.llrp.ltk.generated.parameters.ROSpec;
+import org.llrp.ltk.generated.parameters.ROSpecStartTrigger;
+import org.llrp.ltk.generated.parameters.ROSpecStopTrigger;
+import org.llrp.ltk.generated.parameters.ReaderEventNotificationData;
+import org.llrp.ltk.generated.parameters.ReaderEventNotificationSpec;
+import org.llrp.ltk.generated.parameters.TagReportContentSelector;
+import org.llrp.ltk.generated.parameters.TagReportData;
+import org.llrp.ltk.types.Bit;
+import org.llrp.ltk.types.LLRPMessage;
+import org.llrp.ltk.types.UnsignedByte;
+import org.llrp.ltk.types.UnsignedInteger;
+import org.llrp.ltk.types.UnsignedShort;
+import org.llrp.ltk.types.UnsignedShortArray;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * This LLRP Enterprise Connector uses the Java LLRP Toolkit (llrp.org).
+ * It creates a connection with the reader, sets up an ROSpec (Reader Operation
+ * Specification) to start reading tag data in Field of View (FOV) of the antenna.
+ * It publishes all tag data found in the FOV during a configured duration.
+ * (Default 500 milliseconds).  Some readers may not report the accurate LastSeenTime
+ * and insteads fill it with the FirstSeenTime (use this one for storing).
+ *
+ * Note: Requires an Hostname/IP Address and Port to communicate with the RFID Reader.
+ *
```

Comment:
Assumption:  We end up with one instance of this class for every reader &, thus, for every source?

Reply:
yes, I thought this was a good idea originally :)